scripts/alertmpchanged.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/*
+ * Name: alertmpchanged.php
+ * Description: Mailer for those whose MP has changed at the election
+ */
+
+function mlog($message) {
+    print $message;
+}
+
+include_once dirname(__FILE__) . '/../www/includes/easyparliament/init.php';
+ini_set('memory_limit', -1);
+include_once INCLUDESPATH . 'easyparliament/member.php';
+
+$global_start = getmicrotime();
+$db = new ParlDB;
+
+$nomail = false;
+$onlyemail = '';
+$fromemail = '';
+$fromflag = false;
+$toemail = '';
+$template = 'alert_new_mp';
+for ($k=1; $k<$argc; $k++) {
+	if ($argv[$k] == '--nomail')
+		$nomail = true;
+	if (preg_match('#^--only=(.*)$#', $argv[$k], $m))
+		$onlyemail = $m[1];
+	if (preg_match('#^--from=(.*)$#', $argv[$k], $m))
+		$fromemail = $m[1];
+	if (preg_match('#^--to=(.*)$#', $argv[$k], $m))
+		$toemail = $m[1];
+}
+
+#if (DEVSITE)
+#	$nomail = true;
+
+if ($nomail) mlog("NOT SENDING EMAIL\n");
+if (($fromemail && $onlyemail) || ($toemail && $onlyemail)) {
+	mlog("Can't have both from/to and only!\n");
+	exit;
+}
+
+$active = 0;
+$queries = 0;
+$unregistered = 0;
+$registered = 0;
+$sentemails = 0;
+
+$LIVEALERTS = new ALERT;
+
+$current = array('email' => '', 'token' => '');
+$email_text = array();
+$globalsuccess = 1;
+
+# Fetch all confirmed, non-deleted alerts
+$confirmed = 1; $deleted = 0;
+$alertdata = $LIVEALERTS->fetch($confirmed, $deleted);
+$alertdata = $alertdata['data'];
+
+$outof = count($alertdata);
+$members = array();
+$start_time = time();
+foreach ($alertdata as $alertitem) {
+	$active++;
+	$email = $alertitem['email'];
+    if ($onlyemail && $email != $onlyemail) continue;
+    if ($fromemail && strtolower($email) == $fromemail) $fromflag = true;
+    if ($fromemail && !$fromflag) continue;
+    if ($toemail && strtolower($email) >= $toemail) continue;
+    $criteria_raw = $alertitem['criteria'];
+
+    // we only care about alerts for people speaking
+    if (!strstr($criteria_raw, 'speaker:')) continue;
+
+    preg_match('#speaker:(\d+)#', $criteria_raw, $m);
+    $person_id = $m[1];
+    if (!isset($members[$person_id])) {
+        $queries++;
+        $members[$person_id] = new MEMBER(array('person_id' => $person_id));
+    }
+    $member = $members[$person_id];
+
+    // if they're still elected then don't send the email
+    if ($member->current_member_anywhere()) continue;
+
+    // skip if they didn't lose their westminster seat in the most recent election
+    if ($member->left_house[1]['date'] != '2015-03-30') continue;
```

Comment:
This could use DISSOLUTION_DATE for if it's still around in 5 years (or a few months, who knows).